packages/gatsby-plugin-page-creator/src/gatsby-node.ts

```diff
@@ -41,131 +41,156 @@ export async function createPagesStatefully(
   { path: pagesPath, pathCheck = true, ignore }: IOptions,
   doneCb: PluginCallback
 ): Promise<void> {
-  const { deletePage } = actions
-  const { program } = store.getState()
+  try {
+    const { deletePage } = actions
+    const { program } = store.getState()
 
-  const exts = program.extensions.map(e => `${e.slice(1)}`).join(`,`)
+    const exts = program.extensions.map(e => `${e.slice(1)}`).join(`,`)
 
-  if (!pagesPath) {
-    reporter.panic(`"path" is a required option for gatsby-plugin-page-creator
+    if (!pagesPath) {
+      reporter.panic(`"path" is a required option for gatsby-plugin-page-creator
 
 See docs here - https://www.gatsbyjs.org/plugins/gatsby-plugin-page-creator/`)
-  }
+    }
 
-  // Validate that the path exists.
-  if (pathCheck && !existsSync(pagesPath)) {
-    reporter.panic(`The path passed to gatsby-plugin-page-creator does not exist on your file system:
+    // Validate that the path exists.
+    if (pathCheck && !existsSync(pagesPath)) {
+      reporter.panic(`The path passed to gatsby-plugin-page-creator does not exist on your file system:
 
 ${pagesPath}
 
 Please pick a path to an existing directory.`)
-  }
+    }
 
-  const pagesDirectory = systemPath.resolve(process.cwd(), pagesPath)
-  const pagesGlob = `**/*.{${exts}}`
-
-  // Get initial list of files.
-  let files = await glob(pagesGlob, { cwd: pagesPath })
-  files.forEach(file => {
-    createPage(file, pagesDirectory, actions, ignore, graphql, reporter)
-  })
-
-  watchDirectory(
-    pagesPath,
-    pagesGlob,
-    addedPath => {
-      if (!_.includes(files, addedPath)) {
-        createPage(
-          addedPath,
-          pagesDirectory,
-          actions,
-          ignore,
-          graphql,
-          reporter
-        )
-        files.push(addedPath)
-      }
-    },
-    removedPath => {
-      // Delete the page for the now deleted component.
-      const componentPath = systemPath.join(pagesDirectory, removedPath)
-      store.getState().pages.forEach(page => {
-        if (page.component === componentPath) {
-          deletePage({
-            path: createPath(removedPath),
-            component: componentPath,
+    const pagesDirectory = systemPath.resolve(process.cwd(), pagesPath)
+    const pagesGlob = `**/*.{${exts}}`
+
+    // Get initial list of files.
+    let files = await glob(pagesGlob, { cwd: pagesPath })
+    files.forEach(file => {
+      createPage(file, pagesDirectory, actions, ignore, graphql, reporter)
+    })
+
+    watchDirectory(
+      pagesPath,
+      pagesGlob,
+      addedPath => {
+        try {
+          if (!_.includes(files, addedPath)) {
+            createPage(
+              addedPath,
+              pagesDirectory,
+              actions,
+              ignore,
+              graphql,
+              reporter
+            )
+            files.push(addedPath)
+          }
+        } catch (e) {
+          e.message = `PageCreator: ${e.message}`
+          throw e
```

Comment:
One "catch" is that this rethrow changes the stack trace...

Reply:
hmm good point. hadn't thought of that :\

Reply:
I think it's okay. Just did a few tests and it just appends one function onto it at the bottom. I think it should be okay